cpp/include/cudf/detail/indexalator.cuh

```diff
@@ -56,7 +56,64 @@ namespace detail {
  *  auto result = thrust::find(thrust::device, begin, end, size_type{12} );
  * @endcode
  */
-using input_indexalator = input_normalator<cudf::size_type>;
+struct alignas(16) input_indexalator : base_normalator<input_indexalator, cudf::size_type> {
```

Comment:
Why is CRTP used here? What common functionality is generically invoked by the base class?

Reply:
All of the functions in the base class manipulate the derived member's data pointer.